### PR TITLE
Add price-checking task & e-mail notifications

### PR DESCRIPTION
**Pull request by GeorgePuisha**
Implemented simple scheduler + e-mail framework.

Fix issues with price displaying.

Simple refactoring.

### CODE REVIEW

**Comment by GeorgePuisha** [comment]
Travis-ci is just out of curiosity.

File: server/controllers/price.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+const url = "https://catalog.api.onliner.by/products/";
+
+const findUserById = (id) => models.User.find({
+    where: {
+        id
+    }
+});
+
+const comparePrices = (watched, oldPrice, newPrice) => {
```
Comment (AndreiPiatrou) [inline server/controllers/price.js]: Method name is not so good - it describes only one part of his responsibility. Should be renamed or splitter to two separate methods.

File: server/controllers/price.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const checkForAllWatched = () => {
+    models.Watched.findAll()
+        .then((watchedList) => {
+            watchedList.forEach((watched) => {
+                getPriceByKey(watched.key);
+            });
+        });
+};
+
+module.exports.checkForAllWatched = checkForAllWatched;
```
Comment (AndreiPiatrou) [inline server/controllers/price.js]: Looks like this file is not a controller, but a service. If so it should be moved to `services` folder.

File: server/controllers/price.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+const savePrice = (key, price) => {
+    models.Watched.find({
+        where: {
+            key
+        }
+    }).then((watched) => {
+        addToPrices(watched, price);
+    });
```
Comment (AndreiPiatrou) [inline server/controllers/price.js]: Use sorter version:
```js
then((watched) => addToPrices(watched, price));
```

File: server/controllers/price.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+const getPriceByKey = (key) => {
+    needle.get(url + key, (err, res) => {
+        savePrice(key, res.body.prices.price_min.amount);
+    });
```
Comment (AndreiPiatrou) [inline server/controllers/price.js]: The same here - it can be written without `{}`.

**Pull request merged**